src/modules/commander/state_machine_helper.cpp

```diff
@@ -616,7 +616,7 @@ bool set_nav_state(vehicle_status_s *status, actuator_armed_s *armed, commander_
 			// is not possible and therefore the internal_state needs to be adjusted.
 			internal_state->main_state = commander_state_s::MAIN_STATE_POSCTL;
 
-		} else if (rc_lost && !data_link_loss_act_configured && is_armed) {
+		} else if (rc_lost && status->data_link_lost && !data_link_loss_act_configured && is_armed) {
```

Comment:
That's not what I understand from the comment. What I understand is:

```
if (rc_lost && rc_loss_act_configured && !data_link_loss_act_configured)
```

Reply:
True, the comment was totally off already before. Thanks for noticing.
I changed it to `// Orbit does not depend on RC but while armed & all links lost even when datalink loss is not set up we failsafe` to capture the intent of this case.